src/OVAL/probes/independent/system_info.c

```diff
@@ -334,6 +340,145 @@ static ssize_t __sysinfo_saneval(const char *s)
 	return (ssize_t)real_length;
 }
 
+static char * _offline_chroot_get_menuentry(int entry_num)
+{
+	FILE *fp;
+	pcre *re;
+	const char *error;
+	char *ptr, grubcfg[MAX_BUFFER_SIZE+1] = { '\0' };
+	int erroffset, rc, ovector[30] = { 0 }, len;
```

Comment:
This is a bit of a personal preference but we don't require C89 so you don't have to declare variables at the beginning of scopes. Declaring variables where they are used and limiting their visibility is advised.